src/commands/overlay.ts

```diff
@@ -0,0 +1,72 @@
+import chalk from 'chalk';
+import { mkdir, writeFile } from 'node:fs/promises';
+import { existsSync } from 'node:fs';
+import { dirname } from 'path';
+
+import { API } from '../definition';
+import { confirm as promptConfirm } from '../core/utils/prompts';
+import Command from '../command';
+import * as flagsBuilder from '../flags';
+import { fileArg, overlayFileArg } from '../args';
+import { cli } from '../cli';
+
+export default class Overlay extends Command {
+  static description = 'Apply an OpenAPI specified overlay to your API definition.';
+
+  static examples = [
+    `Apply the OVERLAY_FILE to the existing DEFINITION_FILE. The resulting
+definition is output on stdout meaning you can redirect it to a new
+file.
+
+${chalk.dim('$ bump overlay DEFINITION_FILE OVERLAY_FILE > destination/file.json')}
+* Let's apply the overlay to the main definition... done
+`,
+
+    `You can also specify the output destination file with the
+${chalk.dim('--out')} argument.
+
+${chalk.dim('$ bump overlay DEFINITION_FILE OVERLAY_FILE --out destination/file.json')}
+* Let's apply the overlay to the main definition... done
```

Comment:
I'm not 100% sure we need 2 examples here. Maybe we could only present the --out one?

Reply:
If we keep one, I'd rather teach people the most idiomatic “linux” way of working with CLI (thus the `bump overlay ... > destination/file`)